tools/PI/DevHome.PI/Models/WERDisplayInfo.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+
+namespace DevHome.PI.Models;
+
+public partial class WERDisplayInfo : ObservableObject
+{
+    public WERReport Report { get; }
+
+    [ObservableProperty]
+    private string _failureBucket;
+
+    [ObservableProperty]
+    private string _analyzeResults;
+
+    public WERDisplayInfo(WERReport report)
+    {
+        Report = report;
+        AnalyzeResults = InitializeAnalyzeResults();
+        FailureBucket = UpdateFailureBucket();
+        Report.PropertyChanged += Report_PropertyChanged;
+    }
+
+    private void Report_PropertyChanged(object? sender, PropertyChangedEventArgs e)
+    {
+        if (e.PropertyName == nameof(WERReport.CrashDumpPath))
+        {
+            AnalyzeResults = InitializeAnalyzeResults();
+            FailureBucket = UpdateFailureBucket();
+        }
+        else if (e.PropertyName == nameof(WERReport.FailureBucket))
+        {
+            FailureBucket = UpdateFailureBucket();
+        }
+    }
+
+    private string UpdateFailureBucket()
+    {
+        if (AnalyzeResults == string.Empty)
+        {
+            return Report.FailureBucket;
+        }
+
+        string[] lines = AnalyzeResults.Split('\n', StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Not sure what the content of `AnalyzeResult` is, but would using `Environment.NewLine` work here?

Reply:
The AI suggested using Environment.NewLine first, but in the particular case I was looking at, newlines in the file all ended in \n, but Environment.NewLine was \r\n (or was it \n\r ?). 

Going with just \n covers both cases.